pytools/tag.py

```diff
@@ -362,6 +362,18 @@ class IgnoredForEqualityTag(Tag):
 # }}}
 
 
+# {{{ IgnoredForPropagationTag
+
+class IgnoredForPropagationTag(UniqueTag):
```

Comment:
We should think about the semantics of this a bit more carefully, I think.

![image](https://github.com/inducer/pytools/assets/352067/77ec13fb-98b5-4641-8b05-a69643fd64a8)

There are potentially two "propagation edges" (at least in the pytato context) that could be affected by this, see picture. Which of them is intended to be affected?

We could still represent both types of edges at this level of abstraction, by talking about the "constraint LHS" and "constraint RHS".